spec/support/command_line_helpers.rb

```diff
@@ -17,7 +18,7 @@ def stub_run_for(module_to_stub)
     module_to_stub.stub(:run) do |command, options|
       options ||= {}
       $commands_ran << Hashie::Mash.new(command: command, options: options)
-      command = "" # we need this due to a bug in rspec that will keep this assignment on subsequent runs of the stub
```

Comment:
Double check that the `$commands_ran` isn't globbing the commands together as they get run.  We were having issues before where the `command` was getting appended to on every iteration of the stub.  So two commands getting run (e.g. `git fetch origin` and `git pull origin master`) would cause `$commands_ran` to become `[{command: 'git fetch origin', options: nil}, {command: 'git fetch origingit pull origin master', options: nil}]`